NAVCoin/NavMorph
# Use Prettier across project

This change adds prettier to project.
Prettier will provide automated code formatting. It allows you to not think about manually formatting code for 'easy of reading' and allow `prettier` to do it for you.

You can install a plugin for most IDEs that will format on save for you, or you can just let auto-format when you make a commit.

## Tasks
- [x] Install prettier
- [x] run prettier on the command line across all files
- [x] add pre-commit hook

## Testing
### How to test
- `npm install` in the `home` directory, the `frontend` directory and `server` directory
- `git checkout -b test-pre-commit` to make a branch to run testing
- make a change to any `.ts` or `.js` file, such as adding a `;`
- `git add .` and `git commit -m 'testing'`
- You should see `husky > npm run -s precommit` and see that it changes the file and removes the `;`

### Tests to complete
- [x] Tested on MacOS command line @ro-savage
- [ ] Tested on Windows command line @matt-auckland 
- [ ] Tested when committing from karken @matt-auckland 
- [ ] Tested when committing in vscode @matt-auckland 
- [ ] Tested when committing in webstorm @ro-savage
